AstarteDeviceSDKCSharp/Transport/MQTT/AstarteMqttTransport.cs

```diff
@@ -114,18 +121,30 @@ public override async Task Connect()
                 _client = await InitClientAsync();
             }
 
-            var result = await _client.ConnectAsync(_connectionInfo.GetMqttConnectOptions(),
-                    CancellationToken.None);
-
-            if (result.ResultCode == MqttClientConnectResultCode.Success)
+            try
             {
-                await CompleteAstarteConnection(result.IsSessionPresent);
+                using (var timeoutToken = new CancellationTokenSource(TimeSpan.FromMilliseconds(500)))
```

Comment:
Why did you add a CancellationToken with `500` ms?

Reply:
Mqttnet suggestion is put one second, but i think that is too much and I put on 500 ms.
https://github.com/dotnet/MQTTnet/blob/f79649d71b6544952986bac7f645482617a613ba/Samples/Client/Client_Connection_Samples.cs#L68

Reply:
So if the connection method takes more than 500 ms to connect, the task is cancelled?

Reply:
If device does not connect in 500 ms Cancellationtoken stop Task.

Reply:
The timeout value is too low and cannot be a fixed value, because different networks have different latencies.
I think the best option is to use the same of [MqttClientOptionBuilder]( https://github.com/dotnet/MQTTnet/blob/eb25500f508751365d18ca1e0d7d1b911b1d1c66/Source/MQTTnet/Client/Options/MqttClientOptionsBuilder.cs#L336), leaving the user to modify it via Device constructor.
If it the timeout option is not defined, use the default value of `.net  socket`.